packages/flutter/lib/src/widgets/_web_image_web.dart

```diff
@@ -0,0 +1,241 @@
+    return false;
+  });
+}
+
+/// Returns a widget which displays the [src] in an <img> tag.
+Widget createImgElementWidget(String src,
+        {Key? key,
+        ImageLoadingBuilder? loadingBuilder,
+        ImageFrameBuilder? frameBuilder,
+        ImageErrorWidgetBuilder? errorBuilder}) =>
+    _ImgElementImage(
+      src,
+      key: key,
+      loadingBuilder: loadingBuilder,
+      frameBuilder: frameBuilder,
+      errorBuilder: errorBuilder,
+    );
+
+class _ImgElementImage extends StatefulWidget {
+  const _ImgElementImage(
+    this.src, {
+    super.key,
+    this.loadingBuilder,
+    this.frameBuilder,
+    this.errorBuilder,
+  });
+
+  final String src;
+  final ImageLoadingBuilder? loadingBuilder;
+  final ImageFrameBuilder? frameBuilder;
+  final ImageErrorWidgetBuilder? errorBuilder;
+
+  @override
+  State<StatefulWidget> createState() => _ImgElementImageState();
+}
+
+class _ImgElementImageState extends State<_ImgElementImage> {
+  _ImgElementImageState() {
+    if (_registeredViewType == null) {
+      _register();
+    }
+  }
+
+  // Keeps track if this widget has already registered its view factories.
+  static String? _registeredViewType;
+
+  late _CachedImageInfo _cachedImage;
+
+  void _onCachedImageStateChange() {
+    // Rebuild when the image state changes.
+    setState(() {});
+  }
+
+  static const String _viewType = 'Browser__ImageElementType__';
+
+  static void _register() {
+    assert(_registeredViewType == null);
+    ui_web.platformViewRegistry.registerViewFactory(_viewType, (int viewId,
+        {Object? params}) {
+      final Map<Object?, Object?> paramsMap = params! as Map<Object?, Object?>;
+      final String src = paramsMap['src']! as String;
+      return _WebImageCache._getOrCreateCachedImage(src).image.cloneNode(true);
+    });
+  }
+
+  HtmlElementView? htmlElementView;
+
+  num? naturalWidth;
+  num? naturalHeight;
+
+  int? imgElementId;
+
+  @override
+  void initState() {
+    super.initState();
+    _cachedImage =
+        _WebImageCache._getOrCreateCachedImage(widget.src);
+    _cachedImage.state.addListener(_onCachedImageStateChange);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _cachedImage.state.removeListener(_onCachedImageStateChange);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final _CachedImageInfo info =
+        _WebImageCache._getOrCreateCachedImage(widget.src);
+    final double naturalWidth = info.image.naturalWidth.toDouble();
+    final double naturalHeight = info.image.naturalHeight.toDouble();
+    return ValueListenableBuilder<_ImageLoadingState>(
+      valueListenable: info.state,
+      builder: (BuildContext context, _ImageLoadingState state, Widget? child) {
+        final Widget? builtWidget = switch (state) {
+          _ImageLoadingState.loading =>
+            widget.loadingBuilder?.call(context, child!, null),
+          _ImageLoadingState.success =>
+            widget.frameBuilder?.call(context, child!, 0, true),
```

Comment:
Are we planning on supporting multi-frame images? We might need to mention this in the docs.

Reply:
As far as I know, there is no way to get frame information from an `<img>` tag. I'll add a disclaimer in the WebImage docs.